src/lib/integration_test_cloud_engine/kubernetes_network.ml

```diff
@@ -22,60 +25,79 @@ module Node = struct
 
   let base_kube_args t = [ "--cluster"; t.cluster; "--namespace"; t.namespace ]
 
-  let run_in_postgresql_container node ~cmd =
-    let base_args = base_kube_args node in
-    let base_kube_cmd = "kubectl " ^ String.concat ~sep:" " base_args in
-    let kubectl_cmd =
-      Printf.sprintf "%s -c %s exec -i %s-0 -- %s" base_kube_cmd
-        node.container_id node.container_id cmd
+  (* let run_in_postgresql_container node ~cmd =
+     let base_args = base_kube_args node in
+     let base_kube_cmd = "kubectl " ^ String.concat ~sep:" " base_args in
+     let kubectl_cmd =
+       Printf.sprintf "%s -c %s exec -i %s -- %s" base_kube_cmd
+         node.container_id node.pod_id cmd
+     in
+     let%bind cwd = Unix.getcwd () in
+     Util.run_cmd_exn cwd "sh" [ "-c"; kubectl_cmd ] *)
+
+  let get_logs_in_container ?container node =
+    let container_id : string =
+      match container with None -> node.mina_container_id | Some id -> id
     in
-    let%bind cwd = Unix.getcwd () in
-    Util.run_cmd_exn cwd "sh" [ "-c"; kubectl_cmd ]
-
-  let get_logs_in_container node =
     let base_args = base_kube_args node in
     let base_kube_cmd = "kubectl " ^ String.concat ~sep:" " base_args in
     let pod_cmd =
       sprintf "%s get pod -l \"app=%s\" -o name" base_kube_cmd node.pod_id
     in
-    let%bind cwd = Unix.getcwd () in
+    let%bind.Deferred cwd = Unix.getcwd () in
     let%bind pod = Util.run_cmd_exn cwd "sh" [ "-c"; pod_cmd ] in
     let kubectl_cmd =
-      Printf.sprintf "%s logs -c %s -n %s %s" base_kube_cmd node.container_id
+      Printf.sprintf "%s logs -c %s -n %s %s" base_kube_cmd container_id
         node.namespace pod
     in
     Util.run_cmd_exn cwd "sh" [ "-c"; kubectl_cmd ]
 
-  let run_in_container node cmd =
+  let run_in_container node container_id ~cmd =
     let base_args = base_kube_args node in
     let base_kube_cmd = "kubectl " ^ String.concat ~sep:" " base_args in
     let kubectl_cmd =
       Printf.sprintf
         "%s -c %s exec -i $( %s get pod -l \"app=%s\" -o name) -- %s"
-        base_kube_cmd node.container_id base_kube_cmd node.pod_id cmd
+        base_kube_cmd container_id base_kube_cmd node.pod_id cmd
     in
     let%bind.Deferred cwd = Unix.getcwd () in
-    Malleable_error.return (Util.run_cmd_exn cwd "sh" [ "-c"; kubectl_cmd ])
+    Util.run_cmd_exn cwd "sh" [ "-c"; kubectl_cmd ]
+
+  let run_in_pod node ~cmd =
+    (* run the cmd in every single container in a pod *)
+    let call_run s =
+      (* (run_in_container node s ~cmd) Malleable_error.(>>=) Malleable_error.return Malleable_error.(>>|) Deferred.bind *)
+      let%map res = run_in_container node s ~cmd in
+      res
+    in
+    let foldh accum m =
+      let open Deferred.Let_syntax in
+      let%bind acc = accum in
+      Deferred.map m ~f:(function s -> acc ^ s)
+    in
+    let container_list : string list =
+      List.append [ node.mina_container_id ] []
+      (* (Option.to_list node.mina_archive_container_id) *)
```

Comment:
Nit: Why is this commented out?

Reply:
i suppose we should put that back in.  i commented it out when i was debugging, and realized that when starting, we should only run the /start.sh script in the mina container and not in the archive container, because the archive container is right now not actually running under puppeteer.

Reply:
i changed this, marking resolved